chart/k8gb/templates/dns-endpoint-crd-manifest.yaml

```diff
@@ -1,66 +1,93 @@
 metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.5.0
+    api-approved.kubernetes.io: "https://github.com/kubernetes-sigs/external-dns/pull/2007"
   creationTimestamp: null
-  labels:
-    api: externaldns
-    kubebuilder.k8s.io: 1.0.0
   name: dnsendpoints.externaldns.k8s.io
 spec:
   group: externaldns.k8s.io
   names:
     kind: DNSEndpoint
+    listKind: DNSEndpointList
     plural: dnsendpoints
+    singular: dnsendpoint
   scope: Namespaced
-  subresources:
-    status: {}
-  validation:
-    openAPIV3Schema:
-      properties:
-        apiVersion:
-          type: string
-        kind:
-          type: string
-        metadata:
-          type: object
-        spec:
-          properties:
-            endpoints:
-              items:
-                properties:
-                  dnsName:
-                    type: string
-                  labels:
-                    type: object
-                  providerSpecific:
-                    items:
-                      properties:
-                        name: 
-                          type: string
-                        value:
-                          type: string
-                      type: object
-                    type: array
-                  recordTTL:
-                    format: int64
-                    type: integer
-                  recordType:
-                    type: string
-                  targets:
-                    items:
-                      type: string
-                    type: array
-                type: object
-              type: array
-          type: object
-        status:
-          properties:
-            observedGeneration:
-              format: int64
-              type: integer
-          type: object
-  version: v1alpha1
   versions:
   - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: DNSEndpointSpec defines the desired state of DNSEndpoint
+            properties:
+              endpoints:
+                items:
+                  description: Endpoint is a high-level way of a connection between a service and an IP
+                  properties:
+                    dnsName:
```

Comment:
is this optional field ?

Reply:
@kuritka CRD taken as is from upstream external-dns.  I don't think we can introduce deviation here. Yes, looks like DNSEndpoints aren't validated.